server/lib/related/getMoreOns.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const api = require('next-ft-api-client');
+const articlesToLinks = require('./articlesToLinks');
+const moreOnCount = 3;
+
+const getStreamArticles = metadatum => api.search({
+	filter: ['metadata.idV1', metadatum.idV1],
+
+	// Fetch twice as many as we need, to allow for deduping
+	count: moreOnCount * 2,
+	fields: [
+		'id',
+		'title',
+	],
+}).then(res => res.filter(article => article.title));
+
+
+module.exports = async function getMoreOns(article) {
+	const moreOns = article.metadata.filter(({primary}) => primary);
+	let allLinks = [];
+	const streams = await Promise.all(moreOns.map(getStreamArticles));
+	for(const articles of streams) {
+		allLinks = allLinks.concat(await articlesToLinks(articles));
+	}
+	return allLinks;
```

Comment:
don't look at the compilation though 😱 

![screenshot 2016-04-21 16 16 14](https://cloud.githubusercontent.com/assets/631757/14714303/663e7dd0-07dc-11e6-852f-985b477f1cef.png)